packages/store/src/workspace/workspace.ts

```diff
@@ -58,19 +56,33 @@ export class Workspace {
   }: WorkspaceOptions) {
     this.inlineSuggestionProvider = experimentalInlineSuggestionProvider;
     this._store = new Store(storeOptions);
-    if (storeOptions.blobOptionsGetter) {
-      this._blobOptionsGetter = storeOptions.blobOptionsGetter;
-    }
 
-    if (!storeOptions.isSSR) {
-      this._blobStorage = getBlobStorage(storeOptions.id, k => {
-        return this._blobOptionsGetter ? this._blobOptionsGetter(k) : '';
-      });
-      this._initBlobStorage();
-    } else {
-      // blob storage is not reachable in server side
-      this._blobStorage = Promise.resolve(null);
-    }
+    this._storages = (storeOptions.blobStorages ?? [createMemoryStorage]).map(
+      fn => fn(storeOptions.id)
+    );
+
+    this._blobStorage = {
+      get: async id => {
+        const result = await Promise.allSettled(
+          this._storages.map(s => s.crud.get(id))
+        );
+        const answer = result.find(
+          (result): result is PromiseFulfilledResult<Blob | null> =>
+            result.status === 'fulfilled'
+        );
+        return answer?.value ?? null;
+      },
```

Comment:
May be we can race it here so that the first one settled will be used?

Reply:
No. Because some providers will return `null`. You have to wait for all response and find the first one that is not `null`

Oh, I see my logic problem here

Reply:
something like this. Resolve the result when the first one resolved with a value.

        let found = false;
        let count = 0
        return new Promise(res => {
          this._storages.forEach(s => s.crud.get(id).then(result => {
            if (result.value && !found) {
              found = true;
              res(blob);
            }
            if (++count === this._storages.length && !found) {
              res(null)
            }
          }))
        })

Reply:
thanks